### PR TITLE
OSDOCS-10980: Updating WAF Tutorials

### PR DESCRIPTION
**Pull request by michaelryanmcneill**
Version(s): `enterprise-4.15`+

Issue: https://issues.redhat.com/browse/OSDOCS-10980

Link to docs preview:
https://77808--ocpdocs-pr.netlify.app/openshift-rosa/latest/cloud_experts_tutorials/cloud-experts-using-alb-and-waf.html
https://77808--ocpdocs-pr.netlify.app/openshift-rosa/latest/cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.html

QE review: not required for tutorials

Additional information:

### CODE REVIEW

File: cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc [hunk 1]
```diff
@@ -21,24 +21,21 @@ AWS WAF is a web application firewall that lets you monitor the HTTP and HTTPS r
 
 You can use an AWS Application Load Balancer (ALB) to add a Web Application Firewall (WAF) to your {product-title} (ROSA) workloads. Using an external solution protects ROSA resources from experiencing denial of service due to handling the WAF.
 
-[NOTE]
+[IMPORTANT]
 ====
-It is recommended that you use the xref:../cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.adoc#cloud-experts-using-cloudfront-and-waf[CloudFront method] unless you absolutely must use an ALB based solution.
+It is recommended that you use the significantly more flexible xref:../cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.adoc#cloud-experts-using-cloudfront-and-waf[CloudFront method] unless you absolutely must use an ALB based solution.
```
Comment (jneczypor) [inline cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc:26]:
```suggestion
It is recommended that you use the more flexible xref:../cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.adoc#cloud-experts-using-cloudfront-and-waf[CloudFront method] unless you absolutely must use an ALB based solution.
```

File: cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc [hunk 2]
```diff
@@ -21,24 +21,21 @@ AWS WAF is a web application firewall that lets you monitor the HTTP and HTTPS r
...
 ====
 
-//[Here](https://iamondemand.com/blog/elb-vs-alb-vs-nlb-choosing-the-best-aws-load-balancer-for-your-needs/)'s a good overview of AWS LB types and what they support
-
-// Loosely based off EKS instructions here - https://aws.amazon.com/premiumsupport/knowledge-center/eks-alb-ingress-aws-waf/
-
 [id="prerequisites_{context}"]
 == Prerequisites
 
+* A multi-AZ ROSA (HCP or Classic) cluster.
```
Comment (jneczypor) [inline cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc:32]:
```suggestion
* A multiple availability zone (AZ) ROSA or ROSA (classic architecture) cluster.
```

File: cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc [hunk 3]
```diff
@@ -21,24 +21,21 @@ AWS WAF is a web application firewall that lets you monitor the HTTP and HTTPS r
...
++
 [NOTE]
 ====
-AWS ALBs require a multi-AZ cluster, as well as three public subnets split across three AZs in the same VPC as the cluster.
+AWS Application Load Balancers (ALB) require at least two _public_ subnets across Availability Zones, link:https://docs.aws.amazon.com/elasticloadbalancing/latest/application/application-load-balancers.html#availability-zones[per the AWS documentation]. For this reason, only multi-AZ ROSA clusters can be used with ALBs.
```
Comment (jneczypor) [inline cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc:36]:
```suggestion
AWS ALBs require at least two _public_ subnets across AZs, link:https://docs.aws.amazon.com/elasticloadbalancing/latest/application/application-load-balancers.html#availability-zones[per the AWS documentation]. For this reason, only multiple AZ ROSA clusters can be used with ALBs.
```

File: cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc [hunk 4]
```diff
@@ -71,65 +68,80 @@ This section only applies to clusters that were deployed into existing VPCs. If
...
+  --region ${REGION}
 ----
 
 [id="deploy-aws-load-balancer-operator_{context}"]
 == Deploy the AWS Load Balancer Operator
 
 The link:https://github.com/openshift/aws-load-balancer-operator[AWS Load Balancer Operator] is used to used to install, manage and configure an instance of `aws-load-balancer-controller` in a ROSA cluster. To deploy ALBs in ROSA, we need to first deploy the AWS Load Balancer Operator.
 
-. Create an AWS IAM policy for the AWS Load Balancer Controller:
+. Create a new project to deploy the AWS Load Balancer Operator into:
```
Comment (jneczypor) [inline cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc:114]:
```suggestion
. Create a new project to deploy the AWS Load Balancer Operator into by running the following command:
```

File: cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc [hunk 5]
```diff
@@ -71,65 +68,80 @@ This section only applies to clusters that were deployed into existing VPCs. If
...
++
+[source,terminal]
+----
+$ oc new-project aws-load-balancer-operator
+----
++
+. Create an AWS IAM policy for the AWS Load Balancer Controller if one does not already exist:
```
Comment (jneczypor) [inline cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc:121]:
```suggestion
. Create an AWS IAM policy for the AWS Load Balancer Controller if one does not already exist by running the following command:
```

File: cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc [hunk 6]
```diff
@@ -161,13 +173,17 @@ EOF
...
+$ ROLE_ARN=$(aws iam create-role --role-name "${CLUSTER}-alb-operator" \
    --assume-role-policy-document "file://${SCRATCH}/trust-policy.json" \
    --query Role.Arn --output text)
-$ echo $ROLE_ARN
-
-$ aws iam attach-role-policy --role-name "${CLUSTER_NAME}-alb-operator" \
-     --policy-arn $POLICY_ARN
+----
++
+. Attach the AWS Load Balancer Operator policy to the IAM role we created previously:
```
Comment (jneczypor) [inline cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc:181]:
```suggestion
. Attach the AWS Load Balancer Operator policy to the IAM role we created previously by running the following command:
```

File: cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc [hunk 7]
```diff
@@ -433,11 +449,15 @@ $ curl -X POST "http://${INGRESS}" \
 </html
 ----
 +
+[NOTE]
+====
+Activation of the AWS WAF integration can sometimes take a minute or two. If you do not receive a `403 Forbidden` error, please wait a few seconds and try again.
```
Comment (jneczypor) [inline cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc:454]:
```suggestion
Activation of the AWS WAF integration can sometimes take several minutes. If you do not receive a `403 Forbidden` error, please wait a few seconds and try again.
```

File: cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.adoc [hunk 8]
```diff
@@ -24,7 +24,7 @@ You can use an Amazon CloudFront to add a Web Application Firewall (WAF) to your
 [id="prerequisites_{context}"]
 == Prerequisites
 
-* xref:../rosa_install_access_delete_clusters/rosa-sts-creating-a-cluster-quickly.adoc#rosa-sts-creating-a-cluster-quickly[A ROSA Classic cluster].
+* A ROSA (HCP or Classic) cluster.
```
Comment (jneczypor) [inline cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.adoc:27]:
```suggestion
* A ROSA or ROSA (classic architecture) cluster.
```

File: cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.adoc [hunk 9]
```diff
@@ -35,85 +35,100 @@ You can use an Amazon CloudFront to add a Web Application Firewall (WAF) to your
...
+[source,terminal]
+----
+$ export CLUSTER=my-custom-value
+----
+====
 
-[id="custom_domain_setup{context}"]
-== Custom domain setup
+[id="secondary_ingress_controller_setup_{context}"]
+== Secondary ingress controller setup
```
Comment (jneczypor) [inline cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.adoc:59]:
```suggestion
== Setting up the secondary ingress controller
```

File: cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.adoc [hunk 10]
```diff
@@ -280,33 +294,40 @@ $ aws cloudfront list-distributions --query "DistributionList.Items[?Origins.Ite
...
 [source,text]
 ----
-*.apps.<company_name>.io CNAME d1b2c3d4e5f6g7.cloudfront.net
+*.apps.example.com CNAME d1b2c3d4e5f6g7.cloudfront.net
 ----
 
 [id="deploy-sample-application_{context}"]
 == Deploy a sample application
 
+. Create a new project for your sample application:
```
Comment (jneczypor) [inline cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.adoc:303]:
```suggestion
. Create a new project for your sample application by running the following command:
```

**Review by jneczypor — COMMENTED** [review]
Just a few comments. Mostly for the new ROSA HCP vs. ROSA Classic naming conventions.

File: cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc (hunk 2 above)
Comment (michaelryanmcneill) [reply cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc:32]: I know we're going to be changing the naming, but it hasn't happened across any of the other ones that I published today, so I kept the (HCP or Classic) until we can make all of these changes across the docs.

File: cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.adoc (hunk 8 above)
Comment (michaelryanmcneill) [reply cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.adoc:27]: See above comment about keeping Classic and HCP until we publish that across the docs.

File: cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc [hunk 11]
```diff
@@ -71,65 +68,80 @@ This section only applies to clusters that were deployed into existing VPCs. If
...
 ----
-$ export VPC_ID=<vpc-id>
-$ export PUBLIC_SUBNET_IDS=<public-subnets>
-$ export PRIVATE_SUBNET_IDS=<private-subnets>
+$ export VPC_ID=<vpc-id> <1>
+$ export PUBLIC_SUBNET_IDS=(<space-separated-list-of-ids>) <2>
+$ export PRIVATE_SUBNET_IDS=(<space-separated-list-of-ids>) <3>
 ----
+<1> Replace with the VPC ID of the cluster, for example: `export VPC_ID=vpc-04c429b7dbc4680ba`.
+<2> Replace with a space separated list of the private subnet IDs of the cluster, making sure to preserve the `()`. For example: `export PUBLIC_SUBNET_IDS=(subnet-056fd6861ad332ba2 subnet-08ce3b4ec753fe74c subnet-071aa28228664972f)`.
```
Comment (mburke5678) [inline cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc:76]:
```suggestion
<2> Replace with a space-separated list of the private subnet IDs of the cluster, making sure to preserve the `()`. For example: `export PUBLIC_SUBNET_IDS=(subnet-056fd6861ad332ba2 subnet-08ce3b4ec753fe74c subnet-071aa28228664972f)`.
```

File: cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc [hunk 12]
```diff
@@ -21,24 +21,21 @@ AWS WAF is a web application firewall that lets you monitor the HTTP and HTTPS r
...
 ====
 
-//[Here](https://iamondemand.com/blog/elb-vs-alb-vs-nlb-choosing-the-best-aws-load-balancer-for-your-needs/)'s a good overview of AWS LB types and what they support
-
-// Loosely based off EKS instructions here - https://aws.amazon.com/premiumsupport/knowledge-center/eks-alb-ingress-aws-waf/
-
 [id="prerequisites_{context}"]
 == Prerequisites
 
+* multiple availability zone (AZ) ROSA (HCP or Classic) cluster.
```
Comment (mburke5678) [inline cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc:32]: Should this be

```suggestion
* Multiple availability zone (AZ) ROSA (HCP or Classic) cluster.
```

File: cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc [hunk 13]
```diff
@@ -161,13 +173,17 @@ EOF
...
+$ ROLE_ARN=$(aws iam create-role --role-name "${CLUSTER}-alb-operator" \
    --assume-role-policy-document "file://${SCRATCH}/trust-policy.json" \
    --query Role.Arn --output text)
-$ echo $ROLE_ARN
-
-$ aws iam attach-role-policy --role-name "${CLUSTER_NAME}-alb-operator" \
-     --policy-arn $POLICY_ARN
+----
++
+. Attach the AWS Load Balancer Operator policy to the IAM role we created previously by running the following command:
```
Comment (mburke5678) [inline cloud_experts_tutorials/cloud-experts-using-alb-and-waf.adoc:181]:
```suggestion
. Attach the AWS Load Balancer Operator policy to the IAM role you created previously by running the following command:
```

File: cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.adoc [hunk 14]
```diff
@@ -35,85 +35,100 @@ You can use an Amazon CloudFront to add a Web Application Firewall (WAF) to your
...
+$ export SCRATCH="/tmp/${CLUSTER}/cloudfront-waf"
 $ mkdir -p ${SCRATCH}
-$ echo "Cluster: ${CLUSTER_NAME}, Region: ${REGION}, AWS Account ID: ${AWS_ACCOUNT_ID}"
+$ echo "Cluster: ${CLUSTER}, Region: ${REGION}, AWS Account ID: ${AWS_ACCOUNT_ID}"
 ----
+<1> Replace with the custom domain you want to use for the `IngressController`.
++
+[NOTE]
+====
+The "Cluster" output from the previous command may be the name of your cluster, the internal ID of your cluster, or the cluster's domain prefix. If you prefer to use another identifier, you can manually set this value by running the following command:
```
Comment (mburke5678) [inline cloud_experts_tutorials/cloud-experts-using-cloudfront-and-waf.adoc:51]:
```suggestion
The "Cluster" output from the previous command might be the name of your cluster, the internal ID of your cluster, or the cluster's domain prefix. If you prefer to use another identifier, you can manually set this value by running the following command:
```

**Comment by mburke5678** [comment]
@michaelryanmcneill A few suggestions I was hoping you might consider before we merge. Let me know whe you are ready. 
Otherwise LGTM

**Comment by michaelryanmcneill** [comment]
Completed @mburke5678, can you cherry-pick and merge once these builds run?

**Pull request merged**